snuba/admin/static/nav.tsx

```diff
@@ -1,34 +1,55 @@
-import React from "react";
+import React, { useState } from "react";
 import { COLORS } from "./theme";
 import { NAV_ITEMS } from "./data";
+import Client from "./api_client";
+import { AllowedTools } from "./types";
 
 type NavProps = {
   active: string | null;
   navigate: (nextTab: string) => void;
+  api: Client;
 };
 
 function Nav(props: NavProps) {
-  const { active, navigate } = props;
+  const { active, navigate, api } = props;
+
+  const [allowedTools, setAllowedTools] = useState<string[] | null>(null);
+
+  // Load data if it was not previously loaded
+  if (allowedTools === null) {
+    fetchData();
+  }
+
+  function fetchData() {
+    api.getAllowedTools().then((res) => {
+      setAllowedTools(res.tools);
+    });
+  }
+
   return (
     <nav style={navStyle}>
       <ul style={ulStyle}>
         {NAV_ITEMS.map((item) =>
-          item.id === active ? (
-            <li key={item.id} style={{ color: COLORS.TEXT_DEFAULT }}>
-              <a className="nav-link-active" style={linkStyle}>
-                {item.display}
-              </a>
-            </li>
+          allowedTools?.includes(item.id) || allowedTools?.includes("all") ? (
+            item.id === active ? (
+              <li key={item.id} style={{ color: COLORS.TEXT_DEFAULT }}>
+                <a className="nav-link-active" style={linkStyle}>
+                  {item.display}
+                </a>
+              </li>
+            ) : (
+              <li key={item.id} style={{ color: COLORS.TEXT_INACTIVE }}>
+                <a
+                  className="nav-link"
+                  style={linkStyle}
+                  onClick={() => navigate(item.id)}
+                >
+                  {item.display}
+                </a>
+              </li>
+            )
```

Comment:
Is there any chance we can add a small test for this in the front-end?
Similar to the [test for the kafka page](https://github.com/getsentry/snuba/blob/master/snuba/admin/static/tests/kafka/index.spec.tsx#L21), we can mock the client to return some values. Then do something like:
```typescript
expect(getByTestId('list-item-1')).toHaveStyle(`color: ${COLORS.TEXT_INACTIVE})`)
```

Reply:
Yes! I forgot you added that, I will add a test.